source/core/Synchronization.cpp

```diff
@@ -521,6 +521,7 @@ Boolean QueueCore::PushFront(void* pData)
     IntIndex front = RemoveIndex();
     if (!TryMakeRoom(1, front))
         return false;
+    front = RemoveIndex();  // front may have changed after making room
```

Comment:
Is worth noting that this step is not necessary in the case of `Enqueue` because we are always using the variable `_insert` directly instead of a computed one like `front`.